File: strategy-design-pattern/navigation-go/navigation/navigator.go
package main

type Navigator struct {
	strategy RouteStrategy
}

func NewNavigator(strategy RouteStrategy) *Navigator {
	return &Navigator{strategy: strategy}
}

func (n *Navigator) SetStrategy(strategy RouteStrategy) {
	n.strategy = strategy
}

func (n *Navigator) BuildRoute(start, end Location) RouteResult {
	return n.strategy.BuildRoute(start, end)
}

File: strategy-design-pattern/navigation-python/navigator.py
from models import Location, RouteResult
from strategy import RouteStrategy


class Navigator:
    def __init__(self, strategy: RouteStrategy):
        self._strategy = strategy

    def set_strategy(self, strategy: RouteStrategy) -> None:
        self._strategy = strategy

    def build_route(self, start: Location, end: Location) -> RouteResult:
        return self._strategy.build_route(start, end)
